src/codegate/storage/storage_engine.py

```diff
@@ -152,66 +130,99 @@ async def search(
+                """  # nosec
+                params = [p.lower() for p in packages] + [ecosystem.lower()]
+                logger.debug(
+                    "Searching by package names and ecosystem",
+                    packages=packages,
+                    ecosystem=ecosystem,
+                    sql=query_sql,
+                    params=params,
                 )
-                response.objects = [
-                    obj
-                    for obj in response.objects
-                    if obj.properties["name"].lower() in packages
-                    and obj.properties["type"].lower() == ecosystem.lower()
-                ]
+                cursor.execute(query_sql, params)
+
             elif packages and not ecosystem:
-                response = collection.query.fetch_objects(
-                    filters=wvc.query.Filter.all_of([
-                        wvc.query.Filter.by_property("name").contains_any(packages),
-                    ]),
+                placeholders = ",".join("?" * len(packages))
+                query_sql = f"""
+                    SELECT name, type, status, description
+                    FROM packages
+                    WHERE LOWER(name) IN ({placeholders})
+                """  # nosec
+                params = [p.lower() for p in packages]
+                logger.debug(
+                    "Searching by package names only",
+                    packages=packages,
+                    sql=query_sql,
+                    params=params,
                 )
-                response.objects = [
-                    obj
-                    for obj in response.objects
-                    if obj.properties["name"].lower() in packages
-                ]
+                cursor.execute(query_sql, params)
+
             elif query:
-                # Perform the vector search
-                # Generate the vector for the query
+                # Generate embedding for the query
                 query_vector = await self.inference_engine.embed(self.model_path, [query])
-
-                response = collection.query.near_vector(
-                    query_vector[0],
+                query_embedding = np.array(query_vector[0], dtype=np.float32)
+                query_embedding_bytes = query_embedding.tobytes()
+
+                query_sql = """
+                    WITH distances AS (
+                        SELECT name, type, status, description,
+                               vss_distance(embedding, ?) as distance
+                        FROM packages
+                    )
+                    SELECT name, type, status, description, distance
+                    FROM distances
+                    WHERE distance <= ?
+                    ORDER BY distance ASC
+                    LIMIT ?
+                """  # nosec
+                logger.debug(
+                    "Performing vector similarity search",
+                    query=query,
+                    distance_threshold=distance,
```

Comment:
This query will require a full table scan. It will run slower with larger number of packages in the table. 
Unlike weaviate and other vector dbs, sqlite-vec does not support indexing.
https://alexgarcia.xyz/blog/2024/sqlite-vec-stable-release/index.html#only-brute-force-search-for-now

We will need to watch this and see how well it works with all of the bad packages data.